harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -285,20 +287,28 @@ def _train_for_step(
         num_inputs = 0
 
         for batch_idx in range(start, end):
-            batch = next(self.training_iterator)
+            self.profiler.update_batch_idx(batch_idx)
+            with self.profiler.record_timing("dataloader_next"):
+                batch = next(self.training_iterator)
             num_inputs += pytorch.data_length(batch)
-            batch = self.context.to_device(batch)
+
+            with self.profiler.record_timing("to_device"):
```

Comment:
Maybe "dataloader_to_device"? We can keep the current names for now, but once we look at the webui, we will probably want to think more about what we should be naming the pieces.

Reply:
I'll leave it for now, but totally open to changing all of this.